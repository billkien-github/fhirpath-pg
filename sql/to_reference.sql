SELECT fhirpath_as_reference('{"a":{"b": {"reference": "Patient/1"}}}', '.a.b', 'Reference');
SELECT fhirpath_as_reference('{"a":{"b": {"reference": "http://dom.io/ups/Patient/1"}}}', '.a.b', 'Reference');
