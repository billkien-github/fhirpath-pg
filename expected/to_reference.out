SELECT fhirpath_as_reference('{"a":{"b": {"reference": "Patient/1"}}}', '.a.b', 'Reference');
 fhirpath_as_reference 
-----------------------
 {Patient/1,1}
(1 row)

SELECT fhirpath_as_reference('{"a":{"b": {"reference": "http://dom.io/ups/Patient/1"}}}', '.a.b', 'Reference');
     fhirpath_as_reference     
-------------------------------
 {http://dom.io/ups/Patient/1}
(1 row)

SELECT fhirpath_as_reference('{"a":{"b": {"reference": "Patient/nicola"}}}', '.a.b', 'Reference');
  fhirpath_as_reference  
-------------------------
 {Patient/nicola,nicola}
(1 row)

SELECT fhirpath_as_reference('{"a":{"b": "Patient/1"}}', '.a.b', 'uri');
 fhirpath_as_reference 
-----------------------
 {Patient/1,1}
(1 row)

SELECT fhirpath_as_reference('{"a":{"b": "http://dom.io/ups/Patient/1"}}', '.a.b', 'uri');
     fhirpath_as_reference     
-------------------------------
 {http://dom.io/ups/Patient/1}
(1 row)

SELECT fhirpath_as_reference('{"a":{"b": "Patient/nicola"}}', '.a.b', 'uri');
  fhirpath_as_reference  
-------------------------
 {Patient/nicola,nicola}
(1 row)

SELECT fhirpath_as_reference('{"a":{"b": ["Patient/1", "Patient/2"]}}', '.a.b', 'uri');
   fhirpath_as_reference   
---------------------------
 {Patient/1,1,Patient/2,2}
(1 row)

